File: 프로그래머스/0/181932. 코드 처리하기/코드 처리하기.swift
import Foundation

func solution(_ code:String) -> String {
    var result = ""
    var mode = false
    
    code.enumerated().forEach {
        if $0.element == "1" {
            mode.toggle()
        } else {
            if (mode == false && $0.offset.isMultiple(of: 2)) || (mode == true && !$0.offset.isMultiple(of: 2))  {
                result += String($0.element)
            }
        }
    }
    return result.isEmpty ? "EMPTY" : result
}

File: 프로그래머스/0/181932. 코드 처리하기/코드 처리하기.cpp
#include <string>
#include <vector>

using namespace std;

string solution(string code) {
    string answer = "";
    bool flag = false;

    for (int i = 0; i < code.size(); i++) {
        if (!flag) {
            if (code[i] != '1') {
                if (i % 2 == 0) {
                    answer += code[i];
                }
            } else {
                flag = true;
            }
        } else {
            if (code[i] != '1') {
                if (i % 2 != 0) {
                    answer += code[i];
                }
            } else {
                flag = false;
            }
        }
    }

    return answer.empty() ? "EMPTY" : answer;
}